pyrokinetics/equilibrium.py

```diff
@@ -159,6 +162,173 @@ def read_geqdsk(self, psi_n_lcfs=1.0):
+        """
+
+        Read in TRANSP netCDF and populates Equilibrium object
+
+        """
+
+        import netCDF4 as nc
+        import numpy as np
+        from scipy.interpolate import (
+            RBFInterpolator,
+            InterpolatedUnivariateSpline,
+            RectBivariateSpline,
+        )
+
+        data = nc.Dataset(self.eq_file)
+
+        nr = len(data["XB"][-1, :])
+
+        time_cdf = data["TIME3"][:]
+
+        if isinstance(time, int):
+            itime = time
+        elif isinstance(time, float):
+            itime = np.argmin(abs(time_cdf - time))
+        else:
+            raise ValueError("time input needs to be float or int")
+
+        ntheta = 256
+        theta = np.linspace(0, 2 * np.pi, ntheta)
+        theta = theta[:, np.newaxis]
+
+        # Calculate flux surfaces from moments up to 17
+        R_mom_cos = np.empty((17, ntheta, nr))
+        R_mom_sin = np.empty((17, ntheta, nr))
+        Z_mom_cos = np.empty((17, ntheta, nr))
+        Z_mom_sin = np.empty((17, ntheta, nr))
+
+        for i in range(len(R_mom_cos)):
+            try:
+                R_mom_cos[i, :, :] = (
+                    np.cos(i * theta) * data[f"RMC{i:02d}"][itime, :] * 1e-2
+                )
+            except IndexError:
+                break
+            Z_mom_cos[i, :, :] = (
+                np.cos(i * theta) * data[f"YMC{i:02d}"][itime, :] * 1e-2
+            )
+            if i == 0:
+                R_mom_sin[i, :, :] = 0.0
+                Z_mom_sin[i, :, :] = 0.0
+            else:
+                R_mom_sin[i, :, :] = (
+                    np.sin(i * theta) * data[f"RMS{i:02d}"][itime, :] * 1e-2
+                )
+                Z_mom_sin[i, :, :] = (
+                    np.sin(i * theta) * data[f"YMS{i:02d}"][itime, :] * 1e-2
+                )
+
+        Rsur = np.sum(R_mom_cos, axis=0) + np.sum(R_mom_sin, axis=0)
+        Zsur = np.sum(Z_mom_cos, axis=0) + np.sum(Z_mom_sin, axis=0)
+
+        psi_axis = data["PSI0_TR"][itime]
+        psi_bdry = data["PLFLXA"][itime]
+
+        current = data["PCUR"][itime]
+
+        # Load in 1D profiles
+        q = data["Q"][itime, :]
+        press = data["PMHD_IN"][itime, :]
+
+        # F is on a different grid and need to remove duplicated HFS points
+        psi_rmajm = data["PLFMP"][itime, :]
+        rmajm_ax = np.argmin(psi_rmajm)
+        psi_n_rmajm = psi_rmajm[rmajm_ax:] / psi_rmajm[-1]
+
+        # f = (Bt / |B|) * |B| *  R
+        f = (
+            data["FBTX"][itime, rmajm_ax:]
+            * data["BTX"][itime, rmajm_ax:]
+            * data["RMAJM"][itime, rmajm_ax:]
+            * 1e-2
+        )
+
+        psi = data["PLFLX"][itime, :]
+        psi_n = psi / psi[-1]
+
+        rbdry = Rsur[:, -1]
+        zbdry = Zsur[:, -1]
+
+        Rmin = min(rbdry)
+        Rmax = max(rbdry)
+        Zmin = min(zbdry)
+        Zmax = max(zbdry)
+
+        Rgrid = np.linspace(Rmin, Rmax, nr)
+        Zgrid = np.linspace(Zmin, Zmax, nr)
```

Comment:
Is the Z grid required to have the same size as the R grid?

Reply:
I don't think needs to be the same

Reply:
OK great, in that case you might want to optionally allow user to set nr and nz?

Reply:
Added options in https://github.com/pyro-kinetics/pyrokinetics/pull/48/commits/39f1d67f70d0d8b4190c5b2e90b9de012c76c1bf